Framework/Core/include/Framework/DataAllocator.h

```diff
@@ -133,7 +134,17 @@ class DataAllocator
       void* t2tr = nullptr;
       call_if_defined<struct TreeToTable>([&](auto* p) {
         auto t2t = new std::decay_t<decltype(*p)>(args...);
-        t2t->addAllColumns();
+
+        auto dh = header::DataHeader(spec.description, spec.origin, spec.subSpec);
+        auto colnames = aod::datamodel::getColumnNames(dh);
+        if (colnames.size() == 0) {
+          t2t->addAllColumns();
+        } else {
+          for (auto colname : colnames) {
+            t2t->addColumn(colname.c_str());
+          }
+        }
+
```

Comment:
Why is this needed? The DataAllocator should not know about the DataModel!

Reply:
This is to allow the reader to read only the required branches and not the full tree. To make this more general we discussed we need a table registry.

Reply:
Agreed, but why in the DataAllocator? Can't you do it in the bit which invokes the DataAllocator::make?

Reply:
I can move this to AODReaderHelpers.cxx.

Reply:
done.